.github/workflows/integration.yml

```diff
@@ -1,3 +1,5 @@
+# Run integration tests separately from main.yml because they can be flaky
+# See https://github.com/pypa/twine/issues/684#issuecomment-703150619
```

Comment:
I might follow up on this with a PR that puts the integration tests into `main.yml`, but only running on `push` to `main`.